backend/src/main/kotlin/app/ehrenamtskarte/backend/migration/migrations/V0020_AddApiTokenTable.kt

```diff
@@ -9,13 +9,15 @@ internal class V0020_AddApiTokenTable : Migration() {
             """
                 CREATE TABLE apitokens (
                     id SERIAL PRIMARY KEY ,
-                    "token" BYTEA NOT NULL,
+                    "tokenHash" BYTEA NOT NULL,
                     "creatorId" INTEGER NOT NULL,
                     "projectId" INTEGER NOT NULL,
                     "expirationDate" DATE NOT NULL
                 );
                 ALTER TABLE apitokens ADD CONSTRAINT fk_apitokens_creator__id FOREIGN KEY ("creatorId") REFERENCES administrators(id) ON DELETE RESTRICT ON UPDATE RESTRICT;
                 ALTER TABLE apitokens ADD CONSTRAINT fk_apitokens_projectid__id FOREIGN KEY ("projectId") REFERENCES projects(id) ON DELETE RESTRICT ON UPDATE RESTRICT;
+                
+                ALTER TABLE ONLY apitokens ADD CONSTRAINT apitokens_tokenhash_unique UNIQUE ("tokenHash");
```

Comment:
It seems this migration was already released in version [2024.9.0](https://github.com/digitalfabrik/entitlementcard/blob/2024.9.0-all/backend/src/main/kotlin/app/ehrenamtskarte/backend/migration/migrations/V0020_AddApiTokenTable.kt) ? If so, we should rather make these changes in a new migration? (Maybe I'm missing context though)

Reply:
https://github.com/digitalfabrik/entitlementcard/issues/1645

Reply:
just to ensure here a the current table list
```js
 List of relations
     Schema      |              Name               |   Type   |  Owner
-----------------+---------------------------------+----------+----------
 entitlementcard | acceptingstores                 | table    | backend
 entitlementcard | acceptingstores_id_seq          | sequence | backend
 entitlementcard | addresses                       | table    | backend
 entitlementcard | addresses_id_seq                | sequence | backend
 entitlementcard | administrators                  | table    | backend
 entitlementcard | administrators_id_seq           | sequence | backend
 entitlementcard | applications                    | table    | backend
 entitlementcard | applications_id_seq             | sequence | backend
 entitlementcard | applicationverifications        | table    | backend
 entitlementcard | applicationverifications_id_seq | sequence | backend
 entitlementcard | cards                           | table    | backend
 entitlementcard | cards_id_seq                    | sequence | backend
 entitlementcard | categories                      | table    | backend
 entitlementcard | categories_id_seq               | sequence | backend
 entitlementcard | contacts                        | table    | backend
 entitlementcard | contacts_id_seq                 | sequence | backend
 entitlementcard | geography_columns               | view     | postgres
 entitlementcard | geometry_columns                | view     | postgres
 entitlementcard | migrations                      | table    | backend
 entitlementcard | physicalstores                  | table    | backend
 entitlementcard | physicalstores_id_seq           | sequence | backend
 entitlementcard | projects                        | table    | backend
 entitlementcard | projects_id_seq                 | sequence | backend
 entitlementcard | regions                         | table    | backend
 entitlementcard | regions_id_seq                  | sequence | backend
 entitlementcard | spatial_ref_sys                 | table    | postgres
 entitlementcard | userentitlements                | table    | backend
 entitlementcard | userentitlements_id_seq         | sequence | backend
```

Reply:
I guess for the future, it should usually never be a problem to just add another migration and should usually be preferred over adjusting previous ones, because that is dangerous territory.

Reply:
yes i mean we can can even start here

Reply:
moved to the separate migration